frontend/src/lib/lemon-ui/LemonModal/LemonModal.tsx

```diff
@@ -46,6 +46,11 @@ export interface LemonModalProps {
     contentRef?: React.RefCallback<HTMLDivElement>
     overlayRef?: React.RefCallback<HTMLDivElement>
     'data-attr'?: string
+    /**
+     * some components need more fine control of the z-index
+     * they can push a specific value to control their position in the stacking order
+     */
+    zIndex?: '1061' | '1062' | '1066' | '1067' | '1068' | '1069'
```

Comment:
if we ever do get to the point where a single set of interactions needs more than 5 additional z-indices then we can make the modal component handle this internally on our behalf (or redesign to not stack so much)